app/worker/nats.go

```diff
@@ -0,0 +1,43 @@
+// Copyright 2021 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+package worker
+
+import (
+	"encoding/json"
+
+	"github.com/mendersoftware/workflows/client/nats"
+	"github.com/mendersoftware/workflows/model"
+)
+
+func processNATSTask(natsTask *model.NATSTask, job *model.Job,
+	workflow *model.Workflow, nats nats.Client) (*model.TaskResult, error) {
+
+	var result *model.TaskResult = &model.TaskResult{
+		NATS: &model.TaskResultNATS{},
+	}
+
+	dataJSON := processJobJSON(natsTask.Data, workflow, job)
+	dataJSONBytes, err := json.Marshal(dataJSON)
+	if err == nil {
+		subject := nats.StreamName() + "." + natsTask.Subject
+		err = nats.JetStreamPublish(subject, dataJSONBytes)
+	}
+	result.Success = err == nil
+	if err != nil {
+		result.NATS.Error = err.Error()
+	}
+
+	return result, nil
```

Comment:
I guess you're intentionally not returning the error here since it's probably not going to help attempting to retry at this point.

Reply:
It's `Success == false` which makes the task to be retried, see:
https://github.com/mendersoftware/workflows/blob/master/app/worker/process.go#L73

Return the error stops the execution of the Job. In the case of nats, I thought we want to retry in case of failure.

Reply:
Ahh, I see. It just looks a little confusing since the call signature returns an error but it's always `nil`.